### ISSUE TITLE
TileAt No Longer Returns -1 for Empty Tiles

### ISSUE THREAD

**Issue by SecondDimension** [Scirra/Construct-3-bugs#2940]
## Problem description

When using the tilemap expression TileAt(x, y), the manual states "If the tile at the given position is empty the expression returns -1".  I cannot get this to work, and using  this expression seems to return the last non-empty tileID found.

## Attach a .c3p
[TilemapBug.zip](https://github.com/Scirra/Construct-3-bugs/files/3179452/TilemapBug.zip)

## Steps to reproduce

The attached project has 1 text file and 1 tilemap.  The events simply set the text to show the tile ID at the mouse position.  The right mouse erases a tile.

1. Run the project, and move the cursor over a tile.  You will see the tile ID in the text file
2. Hover over a blank tile, the tile ID remains the same and does not read -1 as stated in the manual
3. Use right mouse to erase a tile under the mouse, this newly erased tile does not show -1 either.

## Observed result

The returned TileID is the last non-empty TileID under the mouse

## Expected result

The returned TileID will show -1 if the tile position in the tilemap is empty

## More details

**Affected browsers/platforms:** Checked in Chrome

**First affected release:** Identified in r149, do not know if previous version are affected

## System details

<details><summary>View details</summary>

Platform information
Browser: NW.js
Browser version: 0.38.0
Browser engine: Chromium
Browser architecture: 64-bit
Context: nwjs
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 4
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (@ac9418ba9c3bd7f6baaffa0b055dfe147e0f8364) NWjs/0.38.0 (like Chrome/74.0.3729.108)
C3 release: r149 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 21 gb
Storage usage (approx): 246 mb (1.1%)
Persistent storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 660 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
It's a mistake in your events. "Cursor is over tilemap" is not true over empty tiles. Change it to "Every tick" to observe it returning -1.

**Comment by SecondDimension** [comment]
Well OK, I see that I messed up in how I setup the example file.  I don't actually use "Cursor is over tilemap" in my project, but wasn't sure how to illustrate the problem in a simple way.  Essentially I have events for autotiling, with no plugins involved, that worked perfectly in C2 and now don't work at all in C3. In fact they crash everything when I run them.  It seems to come down to the fact that my function no longer returns -1 for empty tiles.  It's just frustrating as I have no way to find out why it worked in C2 and not in C3, when I made no changes to it.  I guess I will try the forum again.
